pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/dao/kit/KitDaoImpl.java

```diff
@@ -163,32 +196,98 @@ public Boolean hasTrackingScan(String kitLabel) {
         return booleanCheckFoundAsName(kitLabel, SQL_HAS_KIT_TRACKING);
     }
 
+    /**
+     * Updates scan related information for the kit request.  See
+     * {@link #SET_DDP_KIT_SCAN_INFO_BY_DDP_LABEL_IF_NOT_SET_ALREADY} for exact details
+     * of what gets updated.
+     */
     @Override
-    public Optional<ScanError> updateKitRequest(KitRequestShipping kitRequestShipping, String userId) {
+    public Optional<ScanError> updateKitScanInfo(KitRequestShipping kitRequestShipping, String userId) {
         Optional<ScanError> result = Optional.empty();
         SimpleResult results = inTransaction((conn) -> {
             SimpleResult dbVals = new SimpleResult();
-            try (PreparedStatement stmt = conn.prepareStatement(UPDATE_KIT_REQUEST)) {
+
+            // check to make sure the kit request for the label exists
+            try (PreparedStatement stmt = conn.prepareStatement(KitRequestShipping.SELECT_1_FROM_KIT_REQUEST_WITH_DDP_LABEL)) {
+                stmt.setString(1, kitRequestShipping.getDdpLabel());
+                ResultSet rs = stmt.executeQuery();
+                int numRows = 0;
+                while (rs.next()) {
+                    numRows++;
+                }
+                if (numRows == 0) {
+                    dbVals.resultValue = new ScanError(kitRequestShipping.getDdpLabel(),
+                            "kit request with ddp_label " + kitRequestShipping.getDdpLabel() + " not found.");
+                    return dbVals;
+                } else if (numRows > 1) {
+                    dbVals.resultValue = new ScanError(kitRequestShipping.getDdpLabel(),
+                            String.format("Found %s matching kit request rows for ddp_label %s.",
+                                    numRows, kitRequestShipping.getDdpLabel()));
+                    return dbVals;
+                } // else one row found, as expected
+            } catch (SQLException ex) {
+                logger.error("Not able to query the kit request for ddpLabel " + kitRequestShipping.getDdpLabel(), ex);
+                dbVals.resultValue = new ScanError(kitRequestShipping.getDdpLabel(),
+                        "An unexpected error has occurred");
```

Comment:
Should we return the dbVals here? Or do we want to continue after a SQL exception?